parity/run.rs

```diff
@@ -1002,3 +1024,32 @@ fn wait_for_exit(
 	let _ = exit.1.wait(&mut l);
 	l.clone()
 }
+
+fn wait_for_drop<T>(w: Weak<T>) {
+	let sleep_duration = Duration::from_secs(1);
+	let warn_timeout = Duration::from_secs(30);
+	let max_timeout = Duration::from_secs(120);
+
+	let instant = Instant::now();
+	let mut warned = false;
+
+	loop {
+		{
```

Comment:
Why do we need that additional block?

Reply:
If the `upgrade` is successful is it dropped right away or only at the end of the loop iteration? I just wanted to make sure it was dropped right away (before sleeping).